### PR TITLE
Add implementation of alias in DG

### PR DESCRIPTION
**Pull request by czhi-bin**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,6 +286,23 @@ The following sequence diagrams show how the delete by module code feature works
...
     * Cons: Not able to undo the deletion if the user deletes the wrong batch, it would take a long time to key all the information back in.
 
+### Aliases for different commands:
+
+Aliases are alternative words that you can use to perform the same command. There are a few default aliases provided for every command.
+Users can choose to use their most preferred alias to perform the command they want.
+
+#### Implementation
+
+How the alias work is as such:
```
Comment (jason-ng-zq99) [inline docs/DeveloperGuide.md:296]:
```suggestion
How the alias works:
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,6 +286,23 @@ The following sequence diagrams show how the delete by module code feature works
...
+
+1. When `Logic` is called upon to execute a command, it uses the `ConthacksParser` class to parse the user command.
```
Comment (jason-ng-zq99) [inline docs/DeveloperGuide.md:298]: Will this sound better?
```suggestion
1. When `Logic` is called upon to execute a command, it calls the `ConthacksParser` class to parse the user command.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -283,6 +286,23 @@ The following sequence diagrams show how the delete by module code feature works
...
+2. During the parsing of the user command in `ConthackParser`, `ConthacksParser` separates out the `commandWord` string from the rest of the arguments inside the user command string.
```
Comment (jason-ng-zq99) [inline docs/DeveloperGuide.md:299]:
```suggestion
2. As `ConthackParser` parses the user command, `ConthacksParser` separates the `commandWord` string from the rest of the arguments inside the user command string.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -283,6 +286,23 @@ The following sequence diagrams show how the delete by module code feature works
...
+3. ConthacksParser will pass this `commandWord` string into `CommandWord` enum which contains all the aliases for the different commands, and `CommandWord` class will check if the `commandWord` string is an alias for any of our implemented commands.
```
Comment (jason-ng-zq99) [inline docs/DeveloperGuide.md:300]:
```suggestion
3. `ConthackParser` will pass this `commandWord` string into `CommandWord` enum which contains all the aliases for the different commands, and `CommandWord` class will check if the `commandWord` string is an alias for any of our implemented commands.
```

Comment (jason-ng-zq99) [reply docs/DeveloperGuide.md:300]:
```suggestion
3. `ConthackParser` will pass this `commandWord` string into the `CommandWord` class, which contains all the aliases for the different commands. Then, `CommandWord` class will check if the `commandWord` string is indeed an alias for any of our implemented commands.
```

**Review by jason-ng-zq99 — APPROVED** [review]
Minor changes suggested; LGTM!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (czhi-bin) [inline docs/DeveloperGuide.md:300]:
```suggestion
3. `ConthacksParser` will pass this `commandWord` string into `CommandWord` enum class which contains all the aliases for the different commands. Then `CommandWord` will check if the `commandWord` string is an alias for any of our implemented commands.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -283,6 +286,33 @@ The following sequence diagrams show how the delete by module code feature works
...
+
+Aliases are alternative words that you can use to perform the same command. There are a few default aliases provided for every command.
+Users can choose to use their most preferred alias to perform the command they want.
+
+#### Implementation
+
+How the alias work is as such:
+
+1. When `Logic` is called upon to execute a command, it calls the `ConthacksParser` class to parse the user command.
+2. As `ConthackParser` parses the user command, `ConthacksParser` separates the `commandWord` string from the rest of the arguments inside the user command string.
```
Comment (czhi-bin) [inline docs/DeveloperGuide.md:299]:
```suggestion
2. As `ConthacksParser` parses the user command, `ConthacksParser` separates the `commandWord` string from the rest of the arguments inside the user command string.
```

**Pull request merged**